src/main/kotlin/me/shedaniel/linkie/namespaces/QuiltMappingsNamespace.kt

```diff
@@ -0,0 +1,210 @@
+	val latestQuiltMappingsVersion: String?
+		get() = quiltMappingsBuilds.keys.stream().map { it.tryToVersion() }.filter { it.toString().last().isDigit() && !it.toString().contains("-") }.sorted { a, b -> b!!.compareTo(a!!) }.findFirst().map { it.toString() }.get()
+
+	init {
+		buildSupplier {
+			cached()
+
+			buildVersions {
+				versions { quiltMappingsBuilds.keys }
+				uuid { version ->
+					quiltMappingsBuilds[version]!!.maven.let { it.substring(it.lastIndexOf(':') + 1) }
+				}
+				mappings {
+					MappingsContainer(it, name = "QuiltMappings").apply {
+						loadIntermediaryFromMaven(version)
+						val quiltMappingsMaven = quiltMappingsBuilds[version]!!.maven
+						mappingsSource = loadNamedFromMaven(quiltMappingsMaven.substring(quiltMappingsMaven.lastIndexOf(':') + 1), showError = false)
+					}
+				}
+			}
+		}
+	}
+
+	override fun getDefaultLoadedVersions(): List<String> {
+		return latestQuiltMappingsVersion?.let(::listOf) ?: listOf()
+	}
+
+	override fun getAllVersions(): Sequence<String> = quiltMappingsBuilds.keys.asSequence()
+
+	override fun supportsMixin(): Boolean = true
+	override fun supportsAW(): Boolean = true
+
+	override suspend fun reloadData() {
+		val buildMap = LinkedHashMap<String, MutableList<QuiltMappingsBuild>>()
+
+		val pom = URL("https://maven.quiltmc.org/repository/release/org/quiltmc/quilt-mappings/maven-metadata.xml").readText()
+		SAXReader().read(pom.reader()).rootElement
+				.element("versioning")
+				.element("versions")
+				.elementIterator("version")
+				.asSequence()
+				.map { "org.quitlmc:quilt-mappings:${it.text}" }
+				.forEach {
+					buildMap.computeIfAbsent(it.substring(it.lastIndexOf(":") + 1, it.lastIndexOf("+"))) {
+						mutableListOf()
+					}.add(QuiltMappingsBuild(it))
+				}
+
+		buildMap.forEach { (version, builds) -> builds.maxByOrNull { it.maven.substring(it.maven.lastIndexOf(".") + 1, it.maven.length) }?.apply { quiltMappingsBuilds[version] = this } }
+	}
+
+	override fun getDefaultVersion(channel: () -> String): String =
+			when (channel()) {
+				"patchwork" -> "1.14.4"
+				"snapshot" -> quiltMappingsBuilds.keys.first()
+				else -> latestQuiltMappingsVersion!!
+			}
+
+	override fun getAvailableMappingChannels(): List<String> = listOf(
+			"release",
+			"snapshot",
+			"patchwork",
+	)
+
+	suspend fun MappingsContainer.loadIntermediaryFromMaven(
+			mcVersion: String,
+			repo: String = "https://maven.quiltmc.org/repository/release",
+			group: String = "org.quiltmc.hashed",
+	) =
+			loadIntermediaryFromTinyJar(URL("$repo/${group.replace('.', '/')}/$mcVersion/hashed-$mcVersion.jar"))
+
+	suspend fun MappingsContainer.loadIntermediaryFromTinyJar(url: URL) {
+		url.toAsyncZip().forEachEntry { path, entry ->
+			if (!entry.isDirectory && path.split("/").lastOrNull() == "mappings.tiny") {
+				loadIntermediaryFromTinyInputStream(entry.bytes.inputStream())
+			}
+		}
+	}
+
+	suspend fun MappingsContainer.loadIntermediaryFromTinyInputStream(stream: InputStream) {
```

Comment:
are these all copies of methods in the yarn ns?

Reply:
Yes, I copied the `YarnNamepace` file and removed the unneeded code

Reply:
best to just access and use those code, unprivate any that needs to be public